py/server/deephaven/table_listener.py

```diff
@@ -326,7 +344,7 @@ def stop(self) -> None:
 
 
 def listen(t: Table, listener: Union[Callable[[TableUpdate, bool], None], TableListener], description: str = None, do_replay: bool = False,
```

Comment:
Maybe the right answer here is to insist that users make a `TableListener` if they want to supply `on_error`? Or, require that if they use `TableListener`, they don't separately specify `on_error`?

Reply:
Or could we make the `Callable` option take a `Union` of `TableUpdate` and `Exception`?

Reply:
Having the `Callable` take a `Union[TableUpdate,Exception]` feels like an awkward option.  Think about this case:
1. `Callable` is called with data
2. data causes an exception
3. The same `Callable` is then called again with the exception

One iteration results in 2x calls.  It would feel cleaner to have a separate function.

Reply:
Overall thoughts:

1. `TableListener` and `MergedListener` should both have `on_update` and `on_error` methods.
2. The convenience functions should only accept an `on_error` function if a `Callable` is provided.  Otherwise, the `on_error` in the `*Listener` should be used.
3. The impl is passing around two functions.  This may be ok.  Alternatively, it could just construct a listener from the provided `on_update` plus `on_error`.  Just pointing this out in case it results in a more simple impl.

Reply:
Let me offer one more suggestion that @rcaudy may like and may result in some simplification.  Instead of providing an `on_error` argument, we could accept a tuple of  `on_update` and `on_error` callables for listener.  e.g. `listen(t, (on_update, on_error))`

Then you could define a type like `ListenerLike = Union[Callable[[TableUpdate, bool], None], TableListener, Tuple[Callable[[TableUpdate, bool], None],Callable[[Exception], None]]]`.  We could then use either of the 3 valid input types to construct a listener.

Reply:
> Overall thoughts:
> 
> 1. `TableListener` and `MergedListener` should both have `on_update` and `on_error` methods.
> 2. The convenience functions should only accept an `on_error` function if a `Callable` is provided.  Otherwise, the `on_error` in the `*Listener` should be used.
> 3. The impl is passing around two functions.  This may be ok.  Alternatively, it could just construct a listener from the provided `on_update` plus `on_error`.  Just pointing this out in case it results in a more simple impl.

This is more or less where I arrived with @jmao-denver .

Reply:
> Let me offer one more suggestion that @rcaudy may like and may result in some simplification. Instead of providing an `on_error` argument, we could accept a tuple of `on_update` and `on_error` callables for listener. e.g. `listen(t, (on_update, on_error))`
> 
> Then you could define a type like `ListenerLike = Union[Callable[[TableUpdate, bool], None], TableListener, Tuple[Callable[[TableUpdate, bool], None],Callable[[Exception], None]]]`. We could then use either of the 3 valid input types to construct a listener.

Would that be a breaking change? I certainly like the idea of a tuple (which we then internally make into a `TableListener`). Maybe the only issue is that we don't get to provide a default logging `on_error`, do we?